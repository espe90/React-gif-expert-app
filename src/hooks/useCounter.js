import { useState } from "react";

export default function useCounter(initialcounter=10) {
    const [counter, setCounter] = useState(initialcounter)

    const increment = () => {
        setCounter(counter+1);
    }

    const decrement = () => {
        setCounter(counter-1);
    }

    const reset = () => {
        setCounter(initialcounter);
    }

    return {
        counter,
        increment,
        decrement,
        reset
    }
}
