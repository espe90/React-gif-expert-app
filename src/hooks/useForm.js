import { useState } from "react";

export default function useForm(initialState={}) {
    const [values, setvalues] = useState(initialState)

    const handleInputchange = ({target})=> {
        setvalues({
            ...values,
            [target.name]:target.value
        });
    }

    return [values,handleInputchange];

}
