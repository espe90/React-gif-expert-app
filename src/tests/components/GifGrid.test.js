import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid component', () => {
    let wrapper;
    const category = 'One Punch';
   
    beforeEach(() => {
        useFetchGifs.mockReturnValue({data:[],loading:true});
        wrapper = shallow(<GifGrid category={category} />);
    })

    test('should create component', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show category', () => {
       const cat = wrapper.find('h3').text().trim();
       expect(cat).toBe(category);
    })

    test('should include className animate__fadeIn', () => {
        const cat = wrapper.find('h3');
        expect(cat.prop('className').includes('animate__fadeIn')).toBeTruthy();
    })

    test('should no items loading data', () => {
       expect(wrapper.find('p').exists()).toBe(true);
    })

    test('should no items loading data', () => {
        const gifs = [{
            id:'ABC',
            url:'https://images.jpg',
            title:'Title'
        }];
        useFetchGifs.mockReturnValue({data:gifs,loading:false});
        wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
})
