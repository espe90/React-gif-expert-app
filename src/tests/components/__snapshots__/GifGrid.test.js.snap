// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should create component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing GifGrid component should no items loading data 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Title"
      url="https://images.jpg"
    />
  </div>
</Fragment>
`;
